File: leetcode_C++/p2231.cpp
/*
@Author: mrxhar
@Date:   2023-09-01 10:38:34
@Email:  1337904145@qq.com
@Blog:  https://leetcode.cn/problems/largest-number-after-digit-swaps-by-parity/description/?envType=daily-question&envId=2023-09-01
*/

#include <string>

using namespace std;
class Solution
{
public:
    int largestInteger(int num)
    {
        // 转成字符串
        string s = to_string(num);
        for (int i = 0; i < s.size(); ++i)
        {
            for (int j = i + 1; j < s.size(); ++j)
            {
                if ((s[i] - s[j]) & 1 == 0 && s[i] < s[j])
                {
                    swap(s[i], s[j]);
                }
            }
        }
        return stoi(s);
    }
};

File: leetcode_py/p2231.py
class Solution:
    def largestInteger(self, num: int) -> int:
        # 奇偶 都进行排序
        # 最大值排序
        l = [int(d) for d in list(str(num))]
        n = len(l)

        for i in range(n - 1):
            for j in range(i + 1, n):
                if (l[i] - l[j]) % 2 == 0 and l[i] < l[j]:
                    l[i], l[j] = l[j], l[i]
        return int("".join(str(d) for d in l))
